### PR TITLE
[DTensor] add DTensor sharding strategy for batch norm backward

### LINKED ISSUE #182739: [DTensor] aten.native_batch_norm_backward.default hangs in sharding propagation
I will put up a PR with the fix. 

```
"""Minimal repro: DTensor batch norm backward hangs with channel-dim sharding.

Without a registered strategy for aten.native_batch_norm_backward.default,
the decomposition sharding propagator enters an infinite loop in to_meta()
when inputs are Shard(1).

Usage: torchrun --nproc_per_node=2 repro_batchnorm_hang.py
"""
import torch
import torch.nn.functional as F
from torch.distributed.tensor import Shard, distribute_tensor, init_device_mesh
import torch.distributed as dist

dist.init_process_group("nccl")
torch.cuda.set_device(dist.get_rank())
mesh = init_device_mesh("cuda", (dist.get_world_size(),))

C = dist.get_world_size() * 2
x = distribute_tensor(
    torch.randn(2, C, 3, 3, device="cuda").requires_grad_(True), mesh, [Shard(1)]
)
w = distribute_tensor(torch.randn(C, device="cuda").requires_grad_(True), mesh, [Shard(0)])
b = distribute_tensor(torch.randn(C, device="cuda").requires_grad_(True), mesh, [Shard(0)])
rm = distribute_tensor(torch.zeros(C, device="cuda"), mesh, [Shard(0)])
rv = distribute_tensor(torch.ones(C, device="cuda"), mesh, [Shard(0)])

out = F.batch_norm(x, rm, rv, w, b, training=True)  # forward works
out.sum().backward()  # hangs here
print(f"[rank {dist.get_rank()}] backward done", flush=True)

dist.destroy_process_group()

```

cc @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci @aditvenk @weifengpy @tianyu-l @XilunWu @SherlockNoMad @ppwwyyxx

### PR DESCRIPTION
**Pull request by stmcgovern**
Fixes #182739
Register a single-dim strategy for aten.native_batch_norm_backward.default that enables channel-dim (Shard(1)) sharding. Each channel shard computes its own gradients independently, so no cross-shard communication is needed.

Without this, DTensor's decomposition sharding propagator enters an
infinite loop when resolving cudnn_batch_norm_backward with Shard(1)
inputs, causing the backward pass to hang indefinitely.

The dispatch path is: cudnn_batch_norm_backward (no strategy, decomp
fallback) → native_batch_norm_backward (no strategy, decomp fallback
again) → infinite loop in to_meta(). This strategy intercepts at the
native_batch_norm_backward level, short-circuiting the decomposition.

Authored with Claude.

cc @Skylion007 @ezyang

### CODE REVIEW

**Review by pianpwk — APPROVED** [review]
thanks!

**Review by zpcore — APPROVED** [review]
Thanks for the contribution!
